src/app/services/transaction/pipes/transaction-base.pipe.ts

```diff
@@ -0,0 +1,42 @@
+import { Pipe, PipeTransform } from "@angular/core";
+import { IgxTransactionService, State, Transaction, TransactionType } from "igniteui-angular";
+import { WishlistItem } from "../data";
+
+@Pipe({
+    name: "transactionBasePipe",
+    pure: false
+})
+export class TransactionBasePipe implements PipeTransform {
+    // inject the transaction service
+    constructor(public transactions: IgxTransactionService<Transaction, State>) { }
+
+    public transform(data: WishlistItem[]) {
+        // the pipe should NOT operate on the original dataset
+        // we create a copy of the original data and then use it for visualization only
+        const _data = [...data];
+        const states = this.transactions.getAggregatedChanges(false);
+        // iterate over all pending transactions
+        for (const state of states) {
+            // depending on the type of transaction either:
```

Comment:
depending on the type of the state. You may have add transaction and then update transaction for same id. The type of the state will be add in this case